pepsico/monthly_seasonal_conv_mapping.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import plotly.express as px
+from dash import Dash, dcc, html, Input, Output
+from urllib.request import urlopen
+import json
+
+def compute_annual_monthly_avg(ds, variable, start_year=None, end_year=None):
+    #compute the avg (variable) for each month across the specified years
+    
+    #if precipitation variable, change from kg to mm per day
+    if variable == 'pr':
+        ds[variable] *= 86400
+    
+    if start_year and end_year:
+        ds = ds.sel(T=slice(f"{start_year}-01-01", f"{end_year}-12-31"))
+
+    # Convert daily data to monthly
+    monthly_ds = ds.resample(T="1M").mean()
+
+    # Compute the average for each month across all years
+    monthly_avg = monthly_ds.mean(dim=["X", "Y"])
+    annual_monthly_avg = monthly_avg.groupby("T.month").mean(dim="T")
+    df = annual_monthly_avg.to_dataframe().reset_index()
+    return df
+
+def compute_annual_seasonal_avg(monthly_ds, variable, start_year=None, end_year=None):
+    #compute monthly avgs across all years
+    monthly_avg = monthly_ds.mean(dim=["X", "Y"])
```

Comment:
this comment is misleading: this is doing a spatial average.

Reply:
right, I will change that